### PR TITLE
[global] werf.yaml global refactoring

### PR DESCRIPTION
**Pull request by name212**
## Description
jq update to version 1.7.1;
moving some artifacts to the common module.

Major refactoring of the main build pipeline. Common parts specific to CSE and OSS versions are moved to `.werf/defines/
werf.yaml` is simplified to setting values ​​for the werf template engine and rendering templates from `.werf/werf-*.yaml` templates.
`.werf/werf-*.yaml` templates contain intermediate and final images and define calls from `.werf/defines`, which render the artifact or image completely.

## Why do we need it, and what problem does it solve?
Reducing werf.yaml complexity
Using common build parts in OSS and CSE

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: ci
type: chore
summary: Refactorng werf.yaml.
impact_level: low
---
section: registrypackages
type: chore
summary: Upgrade jq to 1.7.1 version.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by name212** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.29 12517931516 12517931516-1-con-1-29 /sys/deckhouse-oss/install:pr11370 redos@89.169.138.117

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/static'...

File: modules/007-registrypackages/images/jq/werf.inc.yaml [hunk 1]
```diff
@@ -1,9 +1,9 @@
-{{- $version := "1.6" }}
+{{- $version := .CandiVersionMap.jq.version }}
 {{- $image_version := $version | replace "." "-" }}
 ---
 image: {{ $.ModuleName }}/{{ $.ImageName }}-{{ $image_version }}
 from: {{ $.Images.BASE_SCRATCH }}
-git:
+fromCacheVersion: 20241223 # rm if update version
```
Comment (sprait) [inline modules/007-registrypackages/images/jq/werf.inc.yaml:6]: `fromCacheVersion` will not be needed if switch from artifacts to image below

File: .werf/werf-version-map.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+---
+artifact: version-map-artifact
+from: {{ .Images.BASE_ALPINE }}
```
Comment (sprait) [inline .werf/werf-version-map.yaml:3]:
```suggestion
image: version-map-artifact
from: {{ .Images.BASE_ALPINE }}
final: false
```

File: .werf/werf-release-channel.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+---
+image: release-channel-version-prebuild
+from: {{ .Images.BASE_ALPINE_DEV }}
```
Comment (sprait) [inline .werf/werf-release-channel.yaml:3]: should we switch to BASE_ALPINE here?

File: .werf/werf-golang-ci-lint.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+---
+artifact: golangci-lint-artifact
+from: {{ $.Images.BASE_GOLANG_23_ALPINE }}
+shell:
```
Comment (sprait) [inline .werf/werf-golang-ci-lint.yaml:4]:
```suggestion
image: golangci-lint-artifact
from: {{ $.Images.BASE_GOLANG_23_ALPINE }}
final: false
shell:
```

File: .werf/werf-golang-ci-lint.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  beforeInstall:
+  - apk add --no-cache git openssh-client
+  - mkdir -p ~/.ssh && echo "StrictHostKeyChecking accept-new" > ~/.ssh/config
+  install:
+  # Use v1.60.1 for go1.23 - https://github.com/golangci/golangci-lint/issues/4837
+  - export GOPROXY={{ .GOPROXY }}
+  - git clone --depth 1 {{ .SOURCE_REPO }}/golangci/golangci-lint --branch v1.60.1
+  - cd golangci-lint/
+  - CGO_ENABLED=0 GOOS=linux go build -ldflags '-s -w -extldflags "-static"' -o /usr/local/bin/golangci-lint cmd/golangci-lint/main.go
+---
```
Comment (sprait) [inline .werf/werf-golang-ci-lint.yaml:14]:
```suggestion
  - CGO_ENABLED=0 GOOS=linux go build -ldflags '-s -w -extldflags "-static"' -o /usr/local/bin/golangci-lint cmd/golangci-lint/main.go
  - rm -rf /golangci-lint/.git
---
```

File: .werf/defines/base.tmpl [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+  add: /tini/tini-static
+  to: /usr/bin/tini
+  before: setup
+- image: common/semver-artifact
+  add: /usr/local/bin/semver
+  to: /usr/local/bin/semver
+  before: setup
+  {{- range $k8sVersion := $context.k8sVersions }}
+    {{- $image_version := printf "%s.%d" $k8sVersion (index $context.CandiVersionMap "k8s" $k8sVersion "patch") | replace "." "-" }}
+- artifact: common/kubernetes-artifact-{{ $image_version }}
```
Comment (sprait) [inline .werf/defines/base.tmpl:18]:
```suggestion
- image: common/kubernetes-artifact-{{ $image_version }}
```

**Review by sprait — CHANGES_REQUESTED** [review]
many suggestions

File: modules/007-registrypackages/images/jq/werf.inc.yaml (hunk 1 above)
Comment (name212) [reply modules/007-registrypackages/images/jq/werf.inc.yaml]: done

File: .werf/werf-release-channel.yaml (hunk 3 above)
Comment (name212) [reply .werf/werf-release-channel.yaml]: switched

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by sprait — APPROVED** [review]

**Review by RomanenkoDenys — APPROVED** [review]

**Pull request merged**